iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/PhaseTapChangerImpl.java

```diff
@@ -58,6 +58,7 @@ protected NetworkImpl getNetwork() {
     @Override
     public PhaseTapChangerImpl setRegulating(boolean regulating) {
         ValidationUtil.checkPhaseTapChangerRegulation(parent, getRegulationMode(), getRegulationValue(), regulating, getRegulationTerminal(), getNetwork());
+        ValidationUtil.checkTargetDeadband(parent, type, regulating, getTargetDeadband());
```

Comment:
This seems to imply that the target deadband should always be set before using setRegulating. Should we propose a function to set the PST as regulating, together with the definition of the target deadband?

Reply:
It is the same principle as `regulationValue`, `regulationMode` and `regulationTerminal`: `targetDeadband` must be defined before setting regulation status to `true`. To do otherwise, a method with all these parameters should be implemented since they are all dependent on each other and it increases risks of confusion when using the API. The pattern `setAttribute(attribute)` has been chosen to be user-friendly even if the drawback is to be careful of the set order.

Reply:
I am adding some tests to check the exception is thrown if setters are not called in the right order.